### PR TITLE
Help Center: Add current page url to search query mapping

### PR DESCRIPTION
**Pull request by agrullon95**
Related to pdm0RZ-L3-p2

## Proposed Changes

Created a new hook that will map the current page URL to a search query. This will give users relevant search results in the Help Center whenever they navigate to a page that is in the mapping JSON file.

## Testing Instructions

* Checkout branch.
* Sandbox public API and test site.
* Navigate to `cd apps/editing-toolkit` and run `yarn dev --sync`.
* Load Calypso, open the help center, and navigate through all the links to test the different search query mappings.
* Ensure mappings are working for /wp-admin based pages (if specified in JSON).
* Ensure mappings are working in gutenberg editor (if specified in JSON).

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by alshakero — CHANGES_REQUESTED** [review]
Opening the Help Center in my home kills the page.

File: packages/help-center/src/hooks/use-current-page-search-mapping.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import currentPageSearchMapping from '../route-to-query-mapping.json';
+
+export function useCurrentPageSearchMapping( currentRoute: string | undefined ) {
+	let searchQuery = '';
+	if ( currentRoute ) {
+		for ( const key in currentPageSearchMapping ) {
+			if ( currentRoute.startsWith( key ) ) {
+				searchQuery = currentPageSearchMapping[ key as keyof typeof currentPageSearchMapping ];
+				break;
```
Comment (alshakero) [inline packages/help-center/src/hooks/use-current-page-search-mapping.tsx:9]:
```suggestion
				return currentPageSearchMapping[ key as keyof typeof currentPageSearchMapping ];
```

**Review by alshakero — APPROVED** [review]
Tested in wp admin and Calypso. Beautiful and works perfect.

File: packages/help-center/src/route-to-query-mapping.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
+{
+	"/home/": "home",
+	"/stats/": "statistics",
```
Comment (donalirl) [inline packages/help-center/src/route-to-query-mapping.json:3]: Replace current suggestions with: 

https://wordpress.com/support/stats/
https://wordpress.com/support/stats/understand-your-sites-traffic/
https://wordpress.com/support/stats/learn-insights-about-your-website/
https://wordpress.com/support/google-analytics/
https://wordpress.com/support/getting-more-views-and-traffic/

**Pull request merged**